kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,189 @@
+             Spatial support of the padded input.
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8.
+
+        pre_pad is particularly useful when cropping bigger images because
+        this does not introduce border effects inherent to padding.
+
+        """
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False, backend=None):
+        super(Scattering2D_torch, self).__init__(J, shape, max_order=max_order)
+        self.pre_pad, self.L, self.backend = pre_pad, L, backend
+        self.build()
+
+    def build(self):
+        self.M, self.N = self.shape
+        # use the default backend if no backend is provided
+        if not self.backend:
+            from ..backend import torch_backend as backend
+            self.backend = backend
+        elif self.backend.name[0:5] != 'torch':
+            raise RuntimeError('This backend is not supported.')
+
+        if 2 ** self.J > self.shape[0] or 2 ** self.J > self.shape[1]:
+            raise RuntimeError('The smallest dimension should be larger than 2^J.')
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = self.backend.Pad([(self.M_padded - self.M) // 2, (self.M_padded - self.M+1) // 2, (self.N_padded - self.N) // 2,
+                                (self.N_padded - self.N + 1) // 2], [self.M, self.N], pre_pad=self.pre_pad)
+        self.unpad = self.backend.unpad
+        self.create_and_register_filters()
+
+    def create_and_register_filters(self):
+        """ This function run the filterbank function that
+            will create the filters as numpy array, and then, it
+            saves those arrays as module's buffers."""
+
+        # Create the filters
+        filters = filter_bank(self.M_padded, self.N_padded, self.J, self.L)
+        n = 0
+        self.phi, self.psi = filters['phi'], filters['psi']
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c]).unsqueeze(-1) # add a trailing singleton dimension to mark
+                # it as non-complex
+                self.register_buffer('tensor' + str(n), self.phi[c])
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = torch.from_numpy(v).unsqueeze(-1) # add a trailing singleton dimension to mark it
+                    # as non-complex
+                    self.register_buffer('tensor' + str(n), self.psi[j][k])
+                    n += 1
+
+    def scattering(self, input):
+        # each time scattering is run, one needs to make sure self.psi and self.phi point to
+        # the correct buffers
+        n = 0
+        buffer_dict = dict(self.named_buffers())
```

Comment:
ah ok you rely on the exact order. well I guess that works ... but it seems a bit brittle

Reply:
yes, this is a trick. I currently don't have a better idea... buffers are pretty useful.

Reply:
not advocating against the use of buffers. It is essential that you use them so that `.to` and `.cuda` etc work propertly. I was just thinking that instead of calling them `tensor1` `tensor2` etc, they could be called `filter_j2_l4` or something.

I see that the way you do it works reliably.

Having more explicit namings might help people inspect the buffers better, but it is of course not a must

Reply:
my hope is that we will get rid of this mechanism in a future update of torch... so maybe it could be worth not spending much time on it(also observe it's pretty encapuslated in the frontend)

Reply:
get rid of buffers? Seems pretty central to pytorch for me.

As mentioned in my previous comments, I don't think a change is necessary - I just wanted to suggest it

Reply:
Sorry - I meant, in a future release of the PyTorch, I hope we will be able to register a list of tensor without having to register each individual tensor..